File: 프로그래머스/0/120884. 치킨 쿠폰/치킨 쿠폰.js
function solution(chicken) {
    var answer = 0;
    var coupon = chicken;
    
    while (coupon >= 10) {
        var newCoupon = Math.floor(coupon / 10)
        answer += newCoupon;
        coupon = newCoupon + coupon % 10
    }
    
    return answer
}

File: 프로그래머스/0/120884. 치킨 쿠폰/치킨 쿠폰.swift
import Foundation

func solution(_ chicken:Int) -> Int {
    var answer = 0
    var coupon = chicken
    
    while (coupon >= 10) {
        let thisTime = coupon / 10
        answer += thisTime
        coupon = thisTime + coupon % 10
    }
    
    return answer
}